openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -0,0 +1,91 @@
+"""Signal handlers for writing course dates into edx_when."""
+from __future__ import absolute_import, unicode_literals
+
+import logging
+
+from django.dispatch import receiver
+from six import text_type
+from xblock.fields import Scope
+from xmodule.modulestore.django import SignalHandler, modulestore
+
+from edx_when.api import FIELDS_TO_EXTRACT, set_dates_for_course
+
+log = logging.getLogger(__name__)
+
+
+def date_field_values(date_fields, xblock):
+    """
+    Read field values for the specified date fields from the supplied xblock.
+    """
+    result = {}
+    for field_name in date_fields:
+        if field_name not in xblock.fields:
+            continue
+        field = xblock.fields[field_name]
+        if field.scope == Scope.settings and field.is_set_on(xblock):
+            try:
+                result[field.name] = field.read_from(xblock)
+            except TypeError as exception:
+                exception_message = "{message}, Block-location:{location}, Field-name:{field_name}".format(
+                    message=text_type(exception),
+                    location=text_type(xblock.location),
+                    field_name=field.name
+                )
+                raise TypeError(exception_message)
+    return result
+
+
+def extract_dates_from_course(course):
+    """
+    Extract all dates from the supplied course.
+    """
+    log.info('Publishing course dates for %s', course.id)
```

Comment:
This feels like an odd message for this utility method at this point in the flow. You've just logged `Extracting dates from %s` -- maybe move this to right before the `set_dates_for_course` call?

Reply:
Yeah, I didn't look at the logging when I moved this code. I'll do a pass through all of the logging messages.